opendevin/server/listen.py

```diff
@@ -37,6 +45,100 @@
 security_scheme = HTTPBearer()
 
 
+def load_file_upload_config() -> tuple[int, bool, Set[str]]:
+    """
+    Load file upload configuration from the config object.
+
+    This function retrieves the file upload settings from the global config object.
+    It handles the following settings:
+    - Maximum file size for uploads
+    - Whether to restrict file types
+    - List of allowed file extensions
+
+    It also performs sanity checks on the values to ensure they are valid and safe.
+
+    Returns:
+        tuple: A tuple containing:
+            - max_file_size_mb (int): Maximum file size in MB. 0 means no limit.
+            - restrict_file_types (bool): Whether file type restrictions are enabled.
+            - allowed_extensions (set): Set of allowed file extensions.
+    """
+    # Retrieve values from config
+    max_file_size_mb = cast(int, getattr(config, 'file_uploads_max_file_size_mb', 0))
+    restrict_file_types = cast(
+        bool, getattr(config, 'file_uploads_restrict_file_types', False)
+    )
+    allowed_extensions = set(
+        cast(list, getattr(config, 'file_uploads_allowed_extensions', ['.*']))
```

Comment:
Just a small curiosity: is it really necessary to use `getattr` with these? The config object is a dataclass, the fields are defined and have default values. So we should be able to access them. We surely do in other cases. Once the instance is fully initialized, they're there with some value, either our fallback or user-defined. e.g. `config.file_uploads_allowed_extensions` is defined.

Reply:
Here it may well be a little overcautious. I'm still pretty new to Python and I then tend to have extra checks.
Will simplify these lines with a next commit, thanks for highlighting this.